src/__tests__/credit-card-fields-test.tsx

```diff
@@ -0,0 +1,212 @@
+                    ccExp: errorMessage,
+                });
+                expect(onSubmitSpy).not.toHaveBeenCalled();
+            });
+        }
+    }
+);
+
+test('Credit card fields validation, clear expiration date', async () => {
+    render(
+        <ThemeContextProvider theme={makeTheme()}>
+            <Form onSubmit={() => {}}>
+                <CreditCardFields />
+                <ButtonPrimary submit>Submit</ButtonPrimary>
+            </Form>
+        </ThemeContextProvider>
+    );
+
+    const expirationDateField = screen.getByLabelText('Caducidad');
+
+    userEvent.type(screen.getByLabelText('Número de tarjeta'), '4321432143214321');
+    userEvent.type(expirationDateField, '11');
+    userEvent.type(screen.getByLabelText('CVV'), '123');
+
+    expect(screen.getByText('Fecha no válida')).toBeInTheDocument();
+
+    userEvent.clear(expirationDateField);
+    expirationDateField.blur();
+    expect(screen.queryByText('Fecha no válida')).toBeNull();
+    expect(screen.getByText('Este campo es obligatorio')).toBeInTheDocument();
+});
+
+test.each`
+    creditCardNumber      | cvv       | isValid  | description
+    ${'4321432143214321'} | ${'123'}  | ${true}  | ${'3 digits CVV for visa'}
+    ${'5321432143214321'} | ${'123'}  | ${true}  | ${'3 digits CVV for mastercard'}
+    ${'342143214321432'}  | ${'1234'} | ${true}  | ${'4 digits CVV for american express'}
+    ${'4321432143214321'} | ${'12'}   | ${false} | ${'too short CVV for visa'}
+    ${'5321432143214321'} | ${'12'}   | ${false} | ${'too short CVV for mastercard'}
+    ${'342143214321432'}  | ${'123'}  | ${false} | ${'too short CVV for american express'}
+`('Credit card fields validation, $description: $cvv', async ({creditCardNumber, cvv, isValid}) => {
+    const onValidationErrorsSpy = jest.fn();
+    const onSubmitSpy = jest.fn();
+
+    render(
+        <ThemeContextProvider theme={makeTheme()}>
+            <Form onValidationErrors={onValidationErrorsSpy} onSubmit={onSubmitSpy}>
+                <CreditCardFields />
+                <ButtonPrimary submit>Submit</ButtonPrimary>
+            </Form>
+        </ThemeContextProvider>
+    );
+
+    userEvent.type(screen.getByLabelText('Número de tarjeta'), creditCardNumber);
+    userEvent.type(screen.getByLabelText('Caducidad'), '11/50');
+    userEvent.type(screen.getByLabelText('CVV'), cvv);
+    userEvent.click(screen.getByRole('button', {name: 'Submit'}));
+
+    if (isValid) {
+        await waitFor(() => {
+            expect(onSubmitSpy).toHaveBeenCalledWith(
+                {
+                    ccNum: creditCardNumber,
+                    ccExp: {
+                        month: 11,
+                        year: 2050,
+                        raw: '11/50',
+                    },
+                    ccCvv: cvv,
+                },
+                {ccNum: expect.any(String), ccExp: '11/50', ccCvv: cvv}
+            );
+            expect(onValidationErrorsSpy).toHaveBeenCalledWith({});
```

Comment:
not sure if this call is expected here, but the current implementation does it

Reply:
I'd consider that a bug of the implementation. By looking at webapp usages of this prop, no one expects this to be called without errors.

There are two uses:
1. in appointment booking, to scroll into the field with the error (this shouldn't be needed because the form should automatically do this)
2. for analytics, to track form errors (if no errors, nothing is sent)